File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> s = new Stack<Integer>();
        for(int i = 0 ; i < tokens.length ; i++){
            String x = tokens[i];
            if(x.equals("+")){
                int a = s.pop();
                int b = s.pop();
                s.push(a+b);
            }
            else if (x.equals("-")){
                int a = s.pop();
                int b = s.pop();
                s.push(b-a);
            }
            else if (x.equals("*")){
                int a = s.pop();
                int b = s.pop();
                s.push(a*b);
            }
            else if (x.equals("/")){
                int a = s.pop();
                int b = s.pop();
                s.push(b/a);
            }
            else s.push(Integer.parseInt(x));
        }
        return s.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = deque()
        for t in tokens:
            if self.isOperator(t):
                val = self.operation(stack.pop(), stack.pop(), t)
                stack.append(int(val))
            else:
                stack.append(int(t))
        return stack.pop()
    
    @staticmethod
    def isOperator(t: str) -> bool:
        return t in ['+', '-', '*', '/']
    
    @staticmethod
    def operation(second:str, first:str, op: str):
        if op == '+':
            return first + second
        elif op == '-':
            return first - second
        elif op == '/':
            return first / second
        return first * second
        